zebra/zebra_mlag.c

```diff
@@ -78,8 +623,25 @@ DEFPY_HIDDEN (test_mlag,
 			   mlag_role2str(orig, buf1, sizeof(buf1)),
 			   mlag_role2str(orig, buf2, sizeof(buf2)));
 
-	if (orig != zrouter.mlag_info.role)
+	if (orig != zrouter.mlag_info.role) {
 		zsend_capabilities_all_clients();
+		if (zrouter.mlag_info.role != MLAG_ROLE_NONE) {
+			if (zrouter.mlag_info.clients_interested_cnt == 0
+			    && test_mlag_in_progress == false) {
+				if (zrouter.mlag_info.zebra_pth_mlag == NULL)
```

Comment:
I don't think so, the spawning of New MLAG thread always Happens from Main thread.

To avoid corruption, I am posting even the Thread Termination to main thread "zebra_mlag_terminate_pthread"

Reply:
Got your concern, using Mutex.